server/app/controllers/fusor/api/customer_portal/customer_portal_proxies_controller.rb

```diff
@@ -32,6 +32,15 @@ def logout
           render :json => {}
         end
 
+        def is_authenticated
+          authenticated = false
+          if session[:portal_username] && session[:portal_password]
+            # they are not nil, let's make sure neither is empty
+            authenticated = session[:portal_username].empty? && session[:portal_password].empty?
```

Comment:
hrm I must have been on crack when wrote this. WTH. And what's worse is how did I get this to work when I tested it.